src/psbt.js

```diff
@@ -231,6 +252,163 @@ export function psbtOutputFormatter(output) {
   }
 }
 
+/**
+ * Create unchained-wallets style transaction input objects from a PSBT
+ *
+ * @param {module:networks.NETWORKS} network - bitcoin network
+ * @param {String} addressType - address type
+ * @param {Object} psbt - Psbt bitcoinjs-lib object
+ * @return {Object[]} unchained multisig transaction inputs array
+ */
+function getUnchainedInputsFromPSBT(network, addressType, psbt) {
+  return psbt.txInputs.map((input, index) => {
+    const dataInput = psbt.data.inputs[index];
+
+    // FIXME - this is where we're currently only handling P2SH correctly
+    const fundingTxHex = dataInput.nonWitnessUtxo.toString('hex');
+    const fundingTx = Transaction.fromHex(fundingTxHex);
+    const multisig = generateMultisigFromHex(network, addressType, dataInput.redeemScript.toString('hex'));
+
+    return {
+      amountSats: fundingTx.outs[input.index].value,
+      index: input.index,
+      transactionHex: fundingTxHex,
+      txid: reverseBuffer(input.hash).toString('hex'),
+      multisig,
+    }
+  });
+}
+
+/**
+ * Create unchained-wallets style transaction output objects from a PSBT
+ *
+ * @param {Object} psbt - Psbt bitcoinjs-lib object
+ * @return {Object[]} unchained multisig transaction outputs array
+ */
+function getUnchainedOutputsFromPSBT(psbt) {
```

Comment:
Do you think this could just be `getOutputsFromPSBT`? This is very close to the outputs in bitcoinjs too, just renaming the properties slightly, which is fine. I would say that we could even return with the `script` property since [that's in these objects to begin with.](https://github.com/bitcoinjs/bitcoinjs-lib/blob/master/types/psbt.d.ts#L15) and then just call them outputs.